internal/controller/streaming_manager.go

```diff
@@ -42,6 +47,15 @@ type EnvironmentConfig struct {
 }
 
 var resourceTypes = [2]string{"pods", "nodes"}
+var dd = &deadlockDetector{}
+
+// ServerIsHealthy checks if a deadlock has occured within the threaded resource listing process.
+func ServerIsHealthy() bool {
+	if dd.processingResources && time.Since(dd.timeStarted) > 5*time.Minute {
```

Comment:
All accesses to that global variable must be protected by a Mutex.
Otherwise, you won't be able to see updates from this other goroutine.
Add a Mutex to the `deadlockDetector` struct, and `RLock` it here while reading the other fields.